### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by RitzyDevBox**

### CODE REVIEW

**Review by totop716 — CHANGES_REQUESTED** [review]
We still need `tokens.json`?

File: src/constants/index.ts [hunk 1]
```diff
@@ -191,23 +195,172 @@ export const SUPPORTED_WALLETS: { [key: string]: WalletInfo } = {
...
+      ),
+      FRAX: new Token(
+        ChainId.MATIC,
+        '0x45c32fA6DF82ead1e2EF74d17b76547EDdFaFF89',
+        18,
+        'FRAX',
+        'FRAX',
+      ),
+    },
+    UNI: {
```
Comment (totop716) [inline src/constants/index.ts:330]: I think this is not necessary and we can remove this value. We are using this value in the hooks for farms/syrups in `stake/hooks.ts` and `useUniContract` in `hooks/useContract.ts` which is not used and can be removed.

File: src/state/stake/hooks.ts [hunk 2]
```diff
@@ -1083,88 +1095,83 @@ export function useDualStakingInfo(
...
+  return useLairInfo(
+    lairContract,
+    quickContract,
+    lairAddress,
+    quickToken,
+    dQuickToken,
+  );
 }
 
 export function useNewLairInfo(): LairInfo {
```
Comment (sameepsi) [inline src/state/stake/hooks.ts:1114]: Lets keep this function but as I said in the previous comments that we can't showcase new lair in the UI as yet

**Review by sameepsi — CHANGES_REQUESTED** [review]

File: src/components/WalletModal/PendingView.tsx [hunk 3]
```diff
@@ -18,9 +18,11 @@ const PendingView: React.FC<PendingViewProps> = ({
   setPendingError,
   tryActivation,
 }) => {
-  const isMetamask = (window as any).ethereum?.isMetaMask;
-  const isBlockWallet = (window as any).ethereum?.isBlockWallet;
-  const isBitKeep = (window as any).ethereum?.isBitKeep;
+  const { ethereum } = window as any;
+  const isMetamask = ethereum?.isMetaMask;
+  const isBlockWallet = ethereum?.isBlockWallet;
+  const isCypherD = ethereum?.isCypherD;
```
[the remark this answers is not in the document]
Comment (RitzyDevBox) [reply src/components/WalletModal/PendingView.tsx:24]: New Wallet integration which we discussed I created a seperate PR for them ill remove this

File: src/constants/index.ts [hunk 4]
```diff
@@ -191,23 +205,156 @@ export const SUPPORTED_WALLETS: { [key: string]: WalletInfo } = {
...
+      SAND: new Token(
+        ChainId.MATIC,
+        '0xBbba073C31bF03b8ACf7c28EF0738DeCF3695683',
+        18,
+        'SAND',
+        'SAND',
+      ),
+      MAUSDC: new Token(
+        ChainId.MATIC,
+        '0x9719d867A500Ef117cC201206B8ab51e794d3F82',
```
[the remark this answers is not in the document]
Comment (RitzyDevBox) [reply src/constants/index.ts:327]: this address was taken from tokens.json see this line: https://github.com/QuickSwap/interface-v2/blob/24503826d7e27ab0e23df45db96936c274e12893/src/constants/tokens.json#L82

**Review by sameepsi — APPROVED** [review]

**Review by totop716 — APPROVED** [review]
Looks good

**Pull request merged**